### PR TITLE
Handle Further projects

### PR DESCRIPTION
**Pull request by jcapona**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready| [Ticket](https://pi-top.atlassian.net/browse/OS-1272) |

#### Main changes
- Handle Further projects uploaded to the user Desktop, specifically to `~/Desktop/Projects/Further/<username>/`. The project menu page looks for projects in `~/Desktop/Projects/Further/` when entering the `Further` entry in the miniscreen, displaying a selectable list with usernames, which inside contain their projects.
- Refactored `projects.py` into a `projects` module since the file was getting massive. Interesting functions to review that are related to this particular change are `get_folders_with_projects`, `SelectableList.on_state_change` and the `ProjectOverviewList` class.
- Only look for `project.cfg` instead of any `cfg` file for miniscreen projects.

#### Screenshots (feature, test output, profiling, dev tools etc)

[insert screenshots here]

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

File: pt_miniscreen/components/confirmation_page.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def perform_action(self):
             self.on_confirm
         ):
             self.on_confirm()
-            elements_to_pop = 3
+            elements_to_pop = 2
```
Comment (angusjfw) [inline pt_miniscreen/components/confirmation_page.py:82]: a comment to explain this magic number would be good. eg # pop 2 element to get back to top of Projects menu - the project types list and (sometimes?) the project usernames list

File: pt_miniscreen/pages/root/projects/enums.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from enum import auto, Enum
+
+
+class ProjectState(Enum):
+    IDLE = auto()
+    STARTING = auto()
+    RUNNING = auto()
+    PROJECT_USES_MINISCREEN = auto()
```
Comment (angusjfw) [inline pt_miniscreen/pages/root/projects/enums.py:8]: PROJECT_USES_MINISCREEN doesn't sound like a state to me - or at least it's not mutually exclusive with the others

File: pt_miniscreen/pages/root/projects/overview.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            self.update_rows(rows=self.load_rows())
+
+        rows: List[Union[Type[EmptyProjectRow], partial[Row]]] = []
+
+        files = Path(self.directory).glob("*/project.cfg")
+
+        # Sort found files by date/time of last modification
+        for file in sorted(files, key=os.path.getmtime, reverse=True):
+            try:
+                logger.info(f"Trying to read {file}")
```
Comment (angusjfw) [inline pt_miniscreen/pages/root/projects/overview.py:103]: being picky but these should probably be debug logs - this could be logged dozens of times each time this list is loaded, which isn't even really a user action. worth keeping in mind anyway

**Review by angusjfw — COMMENTED** [review]
This is good, I just had some comments about smaller changes which could be clarified.
The refactor of projects is nice, very readable overall. Gutter icons and stuff seems to be working out better than before also.

There is a problem outstanding with the logs. I'm not sure if the refactor has changed this, there were some problems before but I thought we'd resolved them.

From testing it seems to me that both writing and reading the log file can be unreliable. Sometimes I've changed the code and deleted the log file, and the old output still get's written - I wonder whether there is an issue related to reusing or not cleaning up the queue or threads that we use for that.
Other times the log file contents has changed but the viewer shows the old contents. Similarly I wonder if we are reusing the component or it's cache - though this doesn't seem to be the case.

Please try to reproduce these different issues and resolve, we can pair if you like. If you can reproduce the same issues on the master branch then we can go ahead and merge this although we definitely need to sort out the logging before we can introduce people to this feature.

File: pt_miniscreen/pages/root/projects/enums.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+from enum import auto, Enum
+
+
+class ProjectState(Enum):
+    IDLE = auto()
+    STARTING = auto()
+    RUNNING_WITH_INSTRUCTIONS_IN_SCREEN = auto()
+    RUNNING_WITH_CLEAR_CREEN = auto()
```
Comment (angusjfw) [inline pt_miniscreen/pages/root/projects/enums.py:8]: ok this makes more sense but i think typo 🙂 
```suggestion
    RUNNING_WITH_CLEAR_SCREEN = auto()
```

**Review by angusjfw — APPROVED** [review]
LGTM - please create tickets for two non-blocking issues raised above
